src/platform_impl/macos/app_state.rs

```diff
@@ -79,36 +80,38 @@ where
 }
 
 #[derive(Default)]
-struct Handler {
+struct Handler<'a> {
     ready: AtomicBool,
     in_callback: AtomicBool,
     control_flow: Mutex<ControlFlow>,
     control_flow_prev: Mutex<ControlFlow>,
     start_time: Mutex<Option<Instant>>,
     callback: Mutex<Option<Box<dyn EventHandler>>>,
-    pending_events: Mutex<VecDeque<Event<Never>>>,
-    deferred_events: Mutex<VecDeque<Event<Never>>>,
+    pending_events: Mutex<VecDeque<Event<'a, Never>>>,
+    // We don't need it any more. See line 270
+    // deferred_events: Mutex<VecDeque<Event<'a, Never>>>,
```

Comment:
A-ha, now I see.
It eludes me how can we make `WindowEvent::HiDpiChanged` `'static` then.

Reply:
Right now on Windows we just throw away the `HiDpiFactorChanged` event when it can't be dispatched, since the set of circumstances I described is pretty rare, but it occurs to me now that a better solution would be to re-post the OS event at the end of the OS event queue and deal with it later. I'm not entirely sure if that's possible on macOS, so another solution would be to push a new event type to `deferred_events` with no lifetime that gets transformed into a proper `HiDpiFactorChanged` event when it gets dispatched.

Reply:
@Osspial, I went with second option and made a proxy event and a wrapper to pass to `deferred_events`.